src/libraries/System.IO.Pipelines/src/System/IO/Pipelines/StreamPipeWriter.cs

```diff
@@ -256,15 +256,20 @@ public override ValueTask<FlushResult> FlushAsync(CancellationToken cancellation
                 return new ValueTask<FlushResult>(new FlushResult(isCanceled: false, isCompleted: false));
             }
 
-            return FlushAsyncInternal(writeToStream: true, cancellationToken);
+            return FlushAsyncInternal(writeToStream: true, data: Memory<byte>.Empty, cancellationToken);
+        }
+
+        public override ValueTask<FlushResult> WriteAsync(ReadOnlyMemory<byte> source, CancellationToken cancellationToken = default)
+        {
+            return FlushAsyncInternal(writeToStream: true, data: source, cancellationToken);
```

Comment:
Are there situations where this would degrade perf, e.g. if the underlying stream isn't buffered and someone was relying on this instance to do the buffering?

Reply:
`WriteAsync` always flushes so it shouldn't.

Reply:
Actually I guess it *could* cause more writes to the Stream so in a sense it could be worse.

Reply:
Thinking about this, it's easy to work around by appending to the internal buffer.